tests/python/pytorch/dataloading2/test_itemset.py

```diff
@@ -0,0 +1,115 @@
+import dgl
+import torch
+from torch.testing import assert_close
+
+
+def test_ItemSet():
+    # Node or edge IDs.
+    item_set = dgl.ItemSet(torch.arange(0, 5))
+    for i, item in enumerate(item_set):
+        assert i == item.item()
+
+    # Graphs.
+    graphs = [dgl.rand_graph(10, 20) for _ in range(5)]
+    item_set = dgl.ItemSet(graphs)
+    for i, item in enumerate(item_set):
+        assert graphs[i] == item
+
+    # Node pairs.
+    node_pairs = (torch.arange(0, 5), torch.arange(5, 10))
+    item_set = dgl.ItemSet(node_pairs)
+    for i, (src, dst) in enumerate(item_set):
+        assert node_pairs[0][i] == src
+        assert node_pairs[1][i] == dst
+
+    # Node pairs and labels
+    node_pairs = (torch.arange(0, 5), torch.arange(5, 10))
+    labels = torch.randint(0, 3, (5,))
+    item_set = dgl.ItemSet((node_pairs[0], node_pairs[1], labels))
+    for i, (src, dst, label) in enumerate(item_set):
+        assert node_pairs[0][i] == src
+        assert node_pairs[1][i] == dst
+        assert labels[i] == label
+
+    # Head, tail and negative tails.
+    heads = torch.arange(0, 5)
+    tails = torch.arange(5, 10)
+    neg_tails = torch.arange(10, 20).reshape(5, 2)
+    item_set = dgl.ItemSet((heads, tails, neg_tails))
+    for i, (head, tail, negs) in enumerate(item_set):
+        assert heads[i] == head
+        assert tails[i] == tail
+        assert_close(neg_tails[i], negs)
+
+
+def test_DictItemSet():
+    # Node or edge IDs
+    ids = {
+        ("user", "like", "item"): dgl.ItemSet(torch.arange(0, 5)),
+        ("user", "follow", "user"): dgl.ItemSet(torch.arange(5, 10)),
```

Comment:
It is not really related to the test, but usually test should reflect the actual use case.

In our design for hetoro graph, for the train set, should the id of different node all start from zero or continuous?

Reply:
train ids are always required to start from zero? and what about val/test ids which will `ItemSet` as well?

Reply:
I see, correct. But making it continuous [(0,5) then (5, 10)] is a little bit confusing.

Reply:
sure, let's use `(0,5)`